### PR TITLE
Doc: better document --formats (fixes #8315)

### LINKED ISSUE #8315: ogrinfo --formats: document what order is used
$ ogrinfo --formats

Prints the formats in some odd order.

The man page entry should mention what order it is, or mention no order.

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: doc/source/programs/options/formats_vector.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    * A ``v`` is appended for formats supporting virtual IO (``/vsimem``, ``/vsigzip``, ``/vsizip``, etc).
+    * A ``s`` is appended for formats supporting subdatasets.
+
+    The order in which drivers are listed is the one in which they are registered,
+    which determines the order in which they are successively probed when opening
+    a dataset. Most of the time, this order does not matter, but in some situations,
+    several drivers may recognize the same file. The ``-if`` option of some utilities
+    can be specified to ask to use a particular driver to open the dataset.
+    Alternativel, the :config:`GDAL_SKIP` configuration option can also be used
+    to exclude one or several drivers.
```
Comment (jratike80) [inline doc/source/programs/options/formats_vector.rst:18]: Isn't if `OGR_SKIP` for vectors?

Comment (rouault) [reply doc/source/programs/options/formats_vector.rst:18]: > Isn't it `OGR_SKIP` for vectors?

They have been mostly synonymous for a long time. The only difference is the separator (space for GDAL_SKIP, comma for OGR_SKIP)

Comment (jratike80) [reply doc/source/programs/options/formats_vector.rst:18]: Should that reflect homehow to corresponding parts in https://gdal.org/user/configoptions.html?

Comment (rouault) [reply doc/source/programs/options/formats_vector.rst:18]: > Should that reflect homehow to corresponding parts in https://gdal.org/user/configoptions.html?

The :config:`GDAL_SKIP`  will be rendered as a hyperlink to https://gdal.org/user/configoptions.html#driver-management

**Pull request merged**